crates/steel-core/src/primitives/nums.rs

```diff
@@ -84,10 +84,10 @@ fn multiply_2_impl(x: &SteelVal, y: &SteelVal) -> Result<SteelVal> {
             res *= y.as_ref();
             res.into_steelval()
         }
-        (SteelVal::BigFract(x), SteelVal::BigFract(y)) => (x.as_ref() + y.as_ref()).into_steelval(),
+        (SteelVal::BigFract(x), SteelVal::BigFract(y)) => (x.as_ref() * y.as_ref()).into_steelval(),
```

Comment:
Big oops :facepalm:

Reply:
I missed it in the last one, eyes aren't good enough to see it :shrug: